### PR TITLE
Animatable nodes

### PR DESCRIPTION
**Pull request by vicdoval**
#3128

> Right now when the frame changes a global update is performed which is bad idea because many parts of the node tree may not change during the animation.
> I propose to add a is_animatable property to the nodes that can be affected by a scene change, (Objects in, F-Curve, Objects Id selector...) so when the frame change is done the system calls process_from_animatable in stead of a global process.

Added a mixin SvAnimatableNode with a couple of properties:
`is_animatable`: determine if should be called on frame change
`refresh`:  update the node
and a function to draw the icons on the node:
`self.animatable_buttons(layout, icon_only=True)
`
This are the actual Animatable nodes if you miss any please inform:
image

### CODE REVIEW

**Comment by vicdoval** [comment]
@zeffii @Durman  Any thoughts?

**Comment by zeffii** [comment]
nothing constructive at this moment @vicdoval

**Comment by Durman** [comment]
I think there is no need include in the list such nodes which get data from text deitor. Do you suppose to type something their during animation? Keywords for including nodes in the animatable list is is it useful to change it input data during animation. I would suggest to reconsider the list from this point of view. For example I does not see reason to change list of materials of an object.

Why `update the node` property?

**Comment by vicdoval** [comment]
> Why update the node property?

In the `objects in` per example if you disable animation of the node and you make any changes on the scene you need to update the node, and better if it is not a global update 
Before we use to relay on updating every-time everything so when you changed the text of a `Mesh Eval` or `Text input `(with auto re-load) or anything in the scene it was re-evaluated in the next global update. Now you can avoid global updates most of the time so we need a way to update the node to trigger `update_from_node`. 


About the materials i admit is quite strange but it could be the first node of a node-tree and gathers information from the scene (at least one object) so it may be needed, it could also be removed if you wish

the `refresh` property is just a property that calls to updateNode it could be an operator but seemed that like that could make the job without the hassle

**Comment by Durman** [comment]
Why only this nodes should have the `refresh` button?

Probably some nodes can initialized with switched off animatable property then.

**Comment by vicdoval** [comment]
Which other nodes could benefit from the refresh button?

The property is_animatable is a boolean, if both properties where merged you would need to click twice on the is_animatable to refresh and still be independent from the animation system

File: utils/nodes_mixins/sv_animatable_nodes.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    refresh = BoolProperty(
+        name="Update Node",
+        description="Update Node",
+        default=False,
+        update=refresh_node
+    )
+    
+    def animatable_buttons(self, layout, icon_only=False):
+        row = layout.row(align=True)
```
Comment (Durman) [inline utils/nodes_mixins/sv_animatable_nodes.py:50]: We have convention of naming such methods with prefix `draw_` I think.

**Comment by vicdoval** [comment]
This is ready and seems to work nicely if anyone has anything against it say it now or _see you in the issues_

**Pull request merged**